library/ssl_tls.c

```diff
@@ -3744,9 +3745,16 @@ static int ssl_tls13_session_save(const mbedtls_ssl_session *session,
     size_t hostname_len = (session->hostname == NULL) ?
                           0 : strlen(session->hostname) + 1;
 #endif
+
+#if defined(MBEDTLS_SSL_SRV_C) && \
+    defined(MBEDTLS_SSL_EARLY_DATA) && defined(MBEDTLS_SSL_ALPN)
+    const uint8_t alpn_len = (session->ticket_alpn == NULL) ?
+                             0 : (uint8_t) strlen(session->ticket_alpn) + 1;
```

Comment:
This is problematic I think:
1.  (uint8_t) strlen(session->ticket_alpn) + 1 can overflow as `strlen(session->ticket_alpn)` can be up to 255.
2. 255 + 1 length does not fit in one byte

I would suggest rather:
```suggestion
    MBEDTLS_STATIC_ASSERT(MBEDTLS_SSL_MAX_ALPN_NAME_LEN <= 255);
    uint8_t alpn_len = (session->ticket_alpn == NULL) ?
                       0 : (uint8_t) strlen(session->ticket_alpn);
```
and adapt the rest of the code accordingly. One drawback: we cannot use `mbedtls_ssl_session_set_alpn()` in session load.

Reply:
My understanding is that `MBEDTLS_SSL_MAX_ALPN_NAME_LEN` is set to 255 by RFC standard and is not configurable so should we not assume it is left to it's right value and make a check here?